### PR TITLE
Use `fk-metadata` instead of `table-fk-metadata` everywhere

### PR DESCRIPTION
**Pull request by calherries**
In https://github.com/metabase/metabase/issues/38970 we introduced `metabase.driver/describe-fks`, and promised to remove `metabase.driver/describe-table-fks` in 52.

From the changelog:
```
The multimethod metabase.driver/describe-table-fks has been deprecated in favor of metabase.driver/describe-fks. metabase.driver/describe-table-fks will be removed in 0.52.0.
```

This PR preps for that switch by reducing the places `metabase.driver/describe-table-fks` is used to one function that makes the switch easy: `backwards-compatible-describe-fks`

### CODE REVIEW

File: src/metabase/driver.clj [hunk 1]
```diff
@@ -366,7 +366,8 @@
 
 (defmulti describe-fks
   "Returns a reducible collection of maps, each containing information about foreign keys.
-  Takes keyword arguments to narrow down the results to a set of `schema-names` or `table-names`.
+  Takes optional keyword arguments to narrow down the results to a set of `schema-names`
```
Comment (calherries) [inline src/metabase/driver.clj:369]: Just added "optional"

File: src/metabase/sync/sync_metadata/sync_table_privileges.clj [hunk 2]
```diff
@@ -14,10 +14,7 @@
    This is a cache of the data returned from `driver/table-privileges`, but it's stored in the database for performance."
   [database :- (mi/InstanceOf :model/Database)]
   (let [driver (driver.u/database->driver database)]
-    (when (and (not= :redshift driver)
-               ;; redshift does support table-privileges, but we don't want to sync it now because table privileges are
-               ;; meant to enhance action features, but redshift does not support actions for now, so we skip it here.
-               (driver/database-supports? driver :table-privileges database))
+    (when (driver/database-supports? driver :table-privileges database)
```
Comment (calherries) [inline src/metabase/sync/sync_metadata/sync_table_privileges.clj:17]: This is not really related but I just saw it and cleaned it up. Redshift no longer supports table-privileges as of https://github.com/metabase/metabase/pull/41068

File: src/metabase/sync/util.clj [hunk 3]
```diff
@@ -341,11 +339,16 @@
...
+  (t2/reducible-select :model/Table
+                       :db_id (u/the-id database-or-id)
+                       {:where [:and sync-tables-clause
+                                (when (seq schema-names) [:in :schema schema-names])
+                                (when (seq table-names) [:in :name table-names])]}))
 
 (defn db->sync-schemas
-  "Returns all the Schemas that have their metadata sync'd for `database-or-id`."
+  "Returns all the Schemas that have their metadata sync'd for `database-or-id`. If the database doesn't support.
+  Assumes the database supports schemas."
```
Comment (crisptrutski) [inline src/metabase/sync/util.clj:351]: I'm finding this docstring hard to understand with the second sentence fragment, could you reword it to be clearer?

Comment (calherries) [reply src/metabase/sync/util.clj:351]: fixed, thanks

**Review by qnkhuat — APPROVED** [review]
Cool!
